spec/acceptance/provider_validation_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'spec_helper'
+
+require 'open3'
+require 'tempfile'
+
+RSpec.describe 'a provider that returns data that does not match the type schema' do
+  let(:common_args) { '--verbose --trace  --modulepath spec/fixtures' }
+  let(:message) { 'Provider returned data that does not match the Type Schema for `provider_validation' }
+
+  describe 'when strict is default (warning)`' do
+    let(:prefix) { 'Warning:' }
+
+    it 'is raises the expected warnings' do
+      stdout_str, status = Open3.capture2e("puppet resource #{common_args} provider_validation")
```

Comment:
You can re-use the expectations here by wrapping the `describe`s in a loop like this:

```
[ ['warning', 'Warning: '], ['error', 'Error: Could not run:'], ...].each do |strict_value, prefix_value|
describe "when strict is set to #{strict_value}" do
  let(:common_args) { super() + " --strict=#{strict_value}" }
  let(:prefix) { prefix_value }
...
```